### DISCUSSION TITLE
No file prompt when creating new files in the `isowords` local package

### DISCUSSION THREAD

**Discussion by oronbz** [pointfreeco/isowords#150] [Q&A]
**Describe the bug**
When creating new swift file anywhere inside under the `isowords` local package, there's no file name prompt, and a new `File.swift` is instantly created instead. As opposed to creating new file outside the local pacakage, like inside the `AppClip` folder or `Config` folder. I've also noticed that I can't use any of my custom Xcode templates.

I've encountered this on my own projects as well and was wondering if you have a workaround for that, other than creating the file outside the local package and then dragging it inside.

**To reproduce**
1. `open App/isowords.xcodeproj`
2. Right click on the `BottomMenu` folder inside `isowords` ->  `Sources`.
3. Click `New File`
4. Choose `Swift File`
5. Click `Next`

**Expected behavior**
I expect a file name prompt:
![image](https://user-images.githubusercontent.com/1288090/167793799-a8fc6614-f3ba-40eb-ac63-56eaaded4009.png)

**Screenshots/video**
![image](https://user-images.githubusercontent.com/1288090/167793891-9ba31d9f-0c95-4bca-9f00-6bdb0ac9df18.png)

**Environment**
- Xcode Version 13.3.1 (13E500a)

**Comment by mbrandonw** [comment]
This is just how Xcode behaves. Files added to an SPM package skip the file name dialog, and files added to an app target do not. You should consider filing a Feedback with Apple.

I'm going to convert this to a discussion since it is not an issue with the project.

**Reply by oronbz** [reply]
Thanks @mbrandonw  for the reply. I was wondering how you (or any other heavy SPM user) usually handle this issue? Since It can become a real headache with SwiftUI files where you have to rename lots of things or using your own custom file templates.

Do you usually revert to cocoapods workspace? create the files on the target itself and then dragging them inside the package? using tuist instead? Do you think SPM is not production ready yet?

**Reply by rdsquared09** [reply]
@oronbz Resurrecting this one way out of the archive here, after randomly seeing your question 🙂 There's a great description [here](https://samwize.com/2023/02/28/xcode-header-template-for-swift-packages/) about how you can add a new custom file template to Xcode, which will get it to prompt you for a name. Works well!

**Reply by oronbz** [reply]
Haha @rdsquared09 , fortunately, I didn't have to wait for so long to get a proper solution, and found that I could actually use templates. The thing is, I was already using templates before SPM, and they just didn't appear on New File prompt. But later on I've discovered this template property:
```
	<key>SupportsSwiftPackage</key>
	<true/>
```
Which obviously didn't exist on older templates (:
